src/java/grl/src/grl/driver/GRL_Driver.java

```diff
@@ -379,84 +412,48 @@ else if (_currentKUKAiiwaState.armControlState().stateType() == grl.flatbuffer.A
 				if (currentMotion != null && _commandInterface!=grl.flatbuffer.KUKAiiwaInterface.FRI) currentMotion.cancel();
 				if(!cancelTeachMode()) continue;
 
+		        // TODO: switch control mode in config change check? or check in individual armControlState?
```

Comment:
Oh I see, you are saying if the ArmState type is passed with a certain object (TeachArm ShutdownArm MoveArmTrajectory MoveArmJointServo MoveArmCartesianServo) it is possible to determine the desired control mode from the actual command being sent. That makes sense.

I wonder how messy that will get as support for varying EControlMode and EClientCommandMode settings is added. 

Just a design note, [ArmControlState.fbs](https://github.com/ahundt/grl/blob/master/include/grl/flatbuffer/ArmControlState.fbs) was designed so any arm could work with it, then [KUKAiiwa.fbs](https://github.com/ahundt/grl/blob/master/include/grl/flatbuffer/KUKAiiwa.fbs) is supposed to carry the data particular to this robot. If it is pointless we can change that.